hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieErrorTableConfig.java

```diff
@@ -84,6 +84,11 @@ public class HoodieErrorTableConfig extends HoodieConfig {
       .defaultValue(ErrorWriteFailureStrategy.ROLLBACK_COMMIT.name())
       .withDocumentation("The config specifies the failure strategy if error table write fails. "
           + "Use one of - " + Arrays.toString(ErrorWriteFailureStrategy.values()));
+  public static final ConfigProperty<Boolean> ENABLE_ERROR_TABLE_WRITE_UNIFICATION = ConfigProperty
```

Comment:
I am wondering if we really need a flag for this. This seems like it will be more performant for users with the error table writer enabled

Reply:
The error table write implementation can't do a union if they implement the `upsertAndCommit` method so still need to support it behind a feature flag to avoid breaking things for existing users.